src/catalog/pgduckdb_transaction_manager.cpp

```diff
@@ -3,16 +3,15 @@
 
 namespace duckdb {
 
-PostgresTransactionManager::PostgresTransactionManager(AttachedDatabase &db_p, PostgresCatalog &catalog,
-                                                       Snapshot snapshot)
-    : TransactionManager(db_p), catalog(catalog), snapshot(snapshot) {
+PostgresTransactionManager::PostgresTransactionManager(AttachedDatabase &db_p, PostgresCatalog &catalog)
+    : TransactionManager(db_p), catalog(catalog) {
 }
 
 Transaction &
 PostgresTransactionManager::StartTransaction(ClientContext &context) {
-	auto transaction = make_uniq<PostgresTransaction>(*this, context, catalog, snapshot);
-	auto &result = *transaction;
 	lock_guard<mutex> l(transaction_lock);
```

Comment:
Seems like this is probably not the right lock. I think we need to take the global postgres lock.